### PR TITLE
Migrate ACT_PICKUP to the activity actor system

### PR DESCRIPTION
**Pull request by ifreund**
#### Summary
SUMMARY: Infrastructure "Migrate ACT_PICKUP to the activity actor system."

#### Purpose of change
Update legacy activities to use the new system added in #37450.

#### Testing
Ran the test suite locally. Picked up items using the advanced inventory and without.

#### Additional context
I discovered that the move all function of the advanced inventory does, and has never, worked for wearing items, so I added a debug message if it is used and opened #39296.

### CODE REVIEW

**Comment by ifreund** [comment]
rebased to resolve conflicts

**Comment by ifreund** [comment]
Please hold off on merging this until #39646 is merged as this will likely cause conflicts and I'd rather resolve them here.

**Comment by ifreund** [comment]
Conflicts resolved

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -281,6 +282,70 @@ std::unique_ptr<activity_actor> move_items_activity_actor::deserialize( JsonIn &
...
+    // False indicates that the player canceled pickup when met with some prompt
+    const bool keep_going = Pickup::do_pickup( target_items, quantities, autopickup );
+
+    // If there are items left we ran out of moves, so continue the activity
+    // Otherwise, we are done.
+    if( !keep_going || target_items.empty() ) {
+        who.cancel_activity();
+
+        if( who.get_value( "THIEF_MODE_KEEP" ) != "YES" ) {
+            who.set_value( "THIEF_MODE", "THIF_ASK" );
```
Comment (ZhilkinSerg) [inline src/activity_actor.cpp:313]: See https://github.com/CleverRaven/Cataclysm-DDA/issues/40231#issuecomment-625389869

**Pull request merged**